src/main/scala/dotty/dokka/ScalaSignatureProvider.scala

```diff
@@ -0,0 +1,106 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.base.signatures._
+import org.jetbrains.dokka.model._
+import org.jetbrains.dokka.pages._
+import org.jetbrains.dokka.base.signatures.KotlinSignatureProvider
+import org.jetbrains.dokka.base.transformers.pages.comments.CommentsToContentConverter
+import org.jetbrains.dokka.utilities.DokkaLogger
+import collection.JavaConverters._
+import org.jetbrains.dokka.base.translators.documentables._
+import org.jetbrains.dokka.model.properties.PropertyContainer
+import dokka.java.api._
+import java.util.function.Consumer
+import kotlin.jvm.functions.Function2
+import java.util.{List => JList}
+
+class ScalaSignatureProvider(ctcc: CommentsToContentConverter, logger: DokkaLogger) extends SignatureProvider:
+    private val default = new KotlinSignatureProvider(ctcc, logger)
+    private val contentBuilder = new JPageContentBuilder(ctcc, this, logger)
+
+    override def signature(documentable: Documentable) = documentable match {
+        case method: DFunction =>
+           List(methodSignature(method)).asJava
+        case clazz: DClass =>
+           List(classSignature(clazz)).asJava
+        case _ => default.signature(documentable)
+    }
+
+    val styles = Set(TextStyle.Monospace).asJava
+
+    val utils: JvmSignatureUtils = KotlinSignatureUtils.INSTANCE
+
+    private def classSignature(clazz: DClass): ContentNode = 
```

Comment:
If I understand correctly, we're going to model code constructs as `DClass`, `DFunction` etc, correct? I took a look at what data they contain and I think I should be able to use them for link resolution.